internal/transport/transport.go

```diff
@@ -39,10 +40,36 @@ import (
 	"google.golang.org/grpc/tap"
 )
 
+type bufferPool struct {
```

Comment:
A notable downside of this implementation is that it consumes memory indefinitely (200 * 16KB = 3.2MB per connection).  With a `sync.Pool`, excess buffers are freed every GC cycle.  Maybe something can slowly drain this channel to reclaim memory or maybe we should use a `sync.Pool`?  What was the performance of the `sync.Pool` in comparison?

Reply:
I will add benchmark results for both cases, i.e. for sync.Pool and current buffer pool implementation.

Reply:
From the benchmarks posted below, it seems like `sync.Pool` is better anyways. So, I switched to that.